pkg/cdi/cdi.go

```diff
@@ -39,6 +39,10 @@ type CDI interface {
 
 // impl implements CDI interface
 type impl struct {
+	// lastGeneratedName is the last computed file name to which the CDI spec
+	// was written. This is used to remove the spec file when the resource pool
+	// is removed or updated.
+	lastGeneratedName string
```

Comment:
do we need this now ?

Reply:
We create a single file per `resourcePool`. We have to remember that a file has been created, or we need to check if the file exist. First option is fatser.

Reply:
its now going to have the same name, writeSpec will replace the file.
so i dont understand now why we need this bit.

Reply:
If a write fails, we will keep around the old file... It should be deleted first, so we don't keep around irrelevant data right?

Reply:
if write fails, how likely it is for delete to succeed ?
if delete fails, we keep around outdated information right ? so it just moves the possible issue around.

perhaps if CreateCDISpecForPool fails we should fail device plugin altogether.